mysql-test/suite/encryption/t/encrypt_threads_skip_tablespace.test

```diff
@@ -0,0 +1,207 @@
+SET GLOBAL debug="+d,hang_on_ts_with_encryption_key_id_rotation";
+
+SET GLOBAL default_table_encryption=ONLINE_TO_KEYRING;
+# CURRENT_KEY_ID=10 is a marker that we have reached the hang_on_ts_with_encryption_key_id_rotation
+# Can be only one encryption threads - in other case artifical CURRENT_KEY_ID=10 will cause asserts to fail
+SET GLOBAL innodb_encryption_threads=1;
+
+--let $wait_timeout= 600
+--let $wait_condition=SELECT COUNT(*)=1 FROM INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION WHERE CURRENT_KEY_ID=10
+--source include/wait_condition.inc
+
+# Now let's try to exlude ts_encrypted while there are encryption threads running on it.
+# It should fail.
+--error ER_EXCLUDE_ENCRYPTION_THREADS_RUNNING
+ALTER TABLESPACE ts_with_encryption_key_id ENCRYPTION='N'; 
+
+SET GLOBAL debug="-d,hang_on_ts_with_encryption_key_id_rotation";
+
+# Wait for CURRENT_KEY_ID to be restored
+--let $wait_timeout= 600
+--let $wait_condition=SELECT COUNT(*)=0 FROM INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION WHERE CURRENT_KEY_ID=10
+--source include/wait_condition.inc
+
+SET GLOBAL innodb_encryption_threads=4;
+
+insert t1 values (repeat('foobarsecret', 12));
+insert t2 values (repeat('tempsecret', 12));
+insert t3 values (repeat('dummysecret', 12));
+insert t4 values (repeat('verysecret', 12));
+insert t5 values (repeat('moresecret', 12));
+insert t6 values (repeat('sooosecret', 12));
+insert t7 values (repeat('notsosecret', 12));
+insert t8 values (repeat('uuuuusecret', 12));
+
+--let tables_count=`select count(*) from INFORMATION_SCHEMA.INNODB_TABLESPACES`
+--echo # Wait max 10 min for key encryption threads to encrypt all spaces
+--let $wait_timeout= 600
+# - 3 we do not encrypt t3, ts_not_encrypted and temporary tablespace
+--let $wait_condition=SELECT COUNT(*) = $tables_count - 3 FROM INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION WHERE MIN_KEY_VERSION = 1
+--source include/wait_condition.inc
+
+# Make sure are EXLCUDED flags stays the same after encryption
+
+--let $assert_text= TABLE ts3 should have EXCLUDED set to 1 in INNODB_TABLESPACES_ENCRYPTION
+--let $assert_cond= "[SELECT EXCLUDED FROM INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION WHERE name=\\'test/t3\\']" = 1
+--source include/assert.inc
+
+--let $assert_text= TABLE ts_not_encrypted should have EXCLUDED set to 1 in INNODB_TABLESPACES_ENCRYPTION
+--let $assert_cond= "[SELECT EXCLUDED FROM INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION WHERE name=\\'ts_not_encrypted\\']" = 1
+--source include/assert.inc
+
+--let $assert_text=All the other tablespaces should have EXCLUDED set to 0
+--let $assert_cond= "[SELECT MAX(EXCLUDED) FROM INFORMATION_SCHEMA.INNODB_TABLESPACES_ENCRYPTION WHERE name NOT IN(\\'ts_not_encrypted\\',\\'test/t3\\')]" = 0
+--source include/assert.inc
+
+# It should not be possible to EXCLUDE tablespace encrypted by encryption threads
+--error ER_EXPLICIT_DECRYPTION_OF_ONLINE_ENCRYPTED_TABLESPACE
```

Comment:
This looks a limitation imposed. Any reason to not allow?

Reply:
Yes. Because it is online encrypted tablespace user should use encryption threads to decrypt it. Then he can use ENCRYPTION='N' to exclude the tablespace from encryption (encryption by encryption threads). This is how encryption threads were designed. If we want to allow this "offline" decryption and exclusion, we can do that by rebuilding the tablespace. However, this should be done as a separate FR, as this was outside the encryption threads workload.